servers/rendering/renderer_rd/storage_rd/render_scene_data_rd.cpp

```diff
@@ -166,11 +166,16 @@ void RenderSceneDataRD::update_ubo(RID p_uniform_buffer, RS::ViewportDebugDraw p
 		}
 
 		ubo.fog_enabled = render_scene_render->environment_get_fog_enabled(p_env);
+		ubo.fog_mode = render_scene_render->environment_get_fog_mode(p_env);
 		ubo.fog_density = render_scene_render->environment_get_fog_density(p_env);
 		ubo.fog_height = render_scene_render->environment_get_fog_height(p_env);
 		ubo.fog_height_density = render_scene_render->environment_get_fog_height_density(p_env);
 		ubo.fog_aerial_perspective = render_scene_render->environment_get_fog_aerial_perspective(p_env);
 
+		ubo.fog_depth_curve = render_scene_render->environment_get_fog_depth_curve(p_env);
+		ubo.fog_depth_begin = render_scene_render->environment_get_fog_depth_begin(p_env);
+		ubo.fog_depth_end = render_scene_render->environment_get_fog_depth_end(p_env);
```

Comment:
I'd add a check here, like so:
```suggestion
		ubo.fog_depth_end = render_scene_render->environment_get_fog_depth_end(p_env) > 0.0 ? render_scene_render->environment_get_fog_depth_end(p_env) : z_far;
```
And maybe something to ensure the bounds